### PR TITLE
Add command to generate project with cookiecutter

### LINKED ISSUE #1290: Add command to create the file structure for a new Inmanta project
cookie cutter exist

### PR DESCRIPTION
**Pull request by andraskvr**
Closes #1290

### CODE REVIEW

File: src/inmanta/main.py [hunk 1]
```diff
@@ -210,10 +211,25 @@ def project_show(client: Client, project: str) -> None:
...
 
+@project.command(name="generate")
+@click.option("--name", "-n", help="The name of the new project", required=True)
+@click.option("--output-dir", "-o", help="Output directory path")
+@click.pass_obj
+def project_generate(client: Client, name: str, output_dir: Optional[str]) -> None:
+    cookiecutter(
+        "https://github.com/inmanta/inmanta-project-template.git", output_dir=output_dir, extra_context={"project_name": name}
+    )
+    create_project(client, name)
```
Comment (arnaudsjs) [inline src/inmanta/main.py]: In my opinion this command should only create the file structure of an Inmanta project. The project shouldn't be exported to an Inmanta server as well. Or the project creation on the server should at least be optional.

**Comment by arnaudsjs** [comment]
Click shows the docstring of the method when the command is called with `--help`. I think we should include a docstring for the `project create` and the `project generate` commands, since the difference between both commands is not obvious.

File: src/inmanta/main.py [hunk 2]
```diff
@@ -210,10 +211,25 @@ def project_show(client: Client, project: str) -> None:
...
+def create_project(client: Client, name: str) -> None:
     project = client.get_dict("create_project", "project", {"name": name})
     print_table(["Name", "Value"], [["ID", project["id"]], ["Name", project["name"]]])
 
 
+@project.command(name="generate")
+@click.option("--name", "-n", help="The name of the new project", required=True)
+@click.option("--output-dir", "-o", help="Output directory path")
+@click.pass_obj
+def project_generate(client: Client, name: str, output_dir: Optional[str]) -> None:
```
Comment (arnaudsjs) [inline src/inmanta/main.py]: This command fails when:
* `--output-dir` is not set.
* When `output-dir` is set, but the directory already exists.

File: src/inmanta/app.py [hunk 3]
```diff
@@ -293,6 +294,30 @@ def modules(options: argparse.Namespace) -> None:
...
+def generate_project_parser_config(parser: ArgumentParser) -> None:
+    parser.add_argument("--name", "-n", help="The name of the new project", required=True)
+    parser.add_argument("--output-dir", "-o", help="Output directory path", default="./")
+    parser.add_argument(
+        "--default", help="Use default parameters for the project generation", action="store_true", default=False
+    )
+
+
+@command(
+    name="generate-project", help_msg="Generate directory structure for a project", parser_config=generate_project_parser_config
```
Comment (wouterdb) [inline src/inmanta/app.py]: I think this should be a sub command of the project main command. 

@bart: wdyt?

**Pull request merged**